core/src/components/backdrop/backdrop.tsx

```diff
@@ -37,14 +37,22 @@ export class Backdrop implements ComponentInterface {
    */
   @Event() ionBackdropTap!: EventEmitter<void>;
 
-  connectedCallback() {
+  @Element() el!: HTMLElement;
+
+  /** @internal */
+  @Method()
+  async block() {
```

Comment:
> This does change the behavior for using `ion-backdrop` with a custom implementation (outside of Ionic's overlays). Previous to this change, scrolling would be blocked when the backdrop was inserted in the DOM and enabled when removed from the DOM. After this change, that behavior will not occur and the scroll gesture will not be prevented unless the developer uses the internal API methods that our overlay implementation uses.
> 
> I don't consider this a breaking change since it is internal behavior (not an API breakage).

(Threading) I'm confused about this; doesn't this mean that people can no longer block scrolling with `ion-backdrop` if using it as part of a custom implementation? We do have demos for standalone backdrops in the docs -- https://ionicframework.com/docs/api/backdrop -- so I assume this is a supported use case that we would be breaking.

Reply:
The backdrop examples demonstrate blocking click interactions on the content behind, they do not mention anything regarding scroll blocking:

> The backdrop prevents clicking or tapping on the content behind it. 

This change would effect developers that have built custom overlays (not with Ionic), are using `ion-backdrop` within that implementation and are using Ionic's global styles to have `backdrop-no-scroll`. However they could use the internal methods, reflect the classes themselves when they present their overlay or we could decide to make those methods public API as part of a minor release. 

The scroll blocking behavior within `ion-backdrop` is largely specific to Ionic's overlay implementations. To support both use cases, we would need to be modifying the DOM multiple times to disable overflow when the backdrop is inserted into the DOM, enable overflow when the overlay is constructed and then disable overflow when the overlay is presented. I believe this is an unnecessary performance cost for the majority of Ionic use cases.